src/machine.rs

```diff
@@ -261,15 +261,19 @@ impl<'m> Machine<'m> {
     }
 
     async fn send_action(&self, action: Action) -> Result<(), Error> {
+        let vm_id = self.config.vm_id();
         let url: hyper::Uri = Uri::new(&self.config.host_socket_path, "/actions").into();
         let json = serde_json::to_string(&action)?;
+        trace!("{vm_id}: request: {json}");
         let request = Request::builder()
             .method(Method::PUT)
             .uri(url)
             .header("Accept", "application/json")
             .header("Content-Type", "application/json")
             .body(Body::from(json))?;
-        self.client.request(request).await?;
+        let body = self.client.request(request).await?.into_body();
+        let body = hyper::body::to_bytes(body).await?;
```

Comment:
I was thinking that eventually we'll get to the point when we collect and parse the response body to see the possible errors, so it's inevitable.

Debug output for Body is something like "Body (stream)" or something, because it's not collected yet I suppose.

Reply:
> I was thinking that eventually we'll get to the point when we collect and parse the response body to see the possible errors, so it's inevitable.

We wont' be converting it into bytes for that, would we?

Reply:
what would you suggest?

Reply:
I was thinking we'll use [aggregate](https://docs.rs/hyper/latest/hyper/body/fn.aggregate.html) but now I saw the warning in the docs and not sure that'd be great.

However, do we really need to deal with the body in most cases here? Isn't it mostly just some action and we know success from the headers (which gets translated to an Err)?

Reply:
yes, we need to trace only error messages

Reply:
> yes, we need to trace only error messages

then we really don't need to bother with the body.

Reply:
not sure why we don't? if it's a jsonrpc, after checking a status we'll need to get the message out of response, no?

Reply:
I'm confused. We just [agreed above](https://github.com/blockjoy/firec/pull/13#discussion_r887836809) that we don't need to deal with the body.

The reason I said that was because in at least most cases, we don't get anything back except for header as response. In cases where there is a response, we should get it out for sure.

Reply:
I mean we can check the status with `resp.status().is_success()` and in case of error - get the body and trace it